### DISCUSSION TITLE
Pass in a start location offset to the parser as an option

### DISCUSSION THREAD

**Discussion by six5536** [peggyjs/peggy#421] [Ideas]
Sometimes it is useful to parse a set of data in multiple passes, the first splitting the data and subsequent parses parsing each section of data from the first parse. 

We are doing exactly that in our project.

When doing so, it is currently very difficult to keep track of the source locations, because the sub-parse has its source location start from 0 again. All offsets must be muddled in using pretty hacky code.

It would be absolutely great to be able to simple pass in as an option:
```
locationOffset: {
   offset: <startOffset>
   line: <startLine>
   column: <startLocation>
}
```

Where the `locationOffset` is then added to the location returned by the `location()` function (column only applying to the first line).

I'm happy to create a PR for this.

**Comment by hildjj** [comment]
I like this idea a lot.  The relatively-new `GrammarLocation` does about 80% of what you want, when passed in as a grammarSource.  It assumes that you've sliced the original text down to just the input, but does everything else you want, I think.

If we're going to do something more with GrammarLocation, we'll also need to pass in a length, since your input is likely to be a substring of a larger block of text.

**Reply by six5536** [reply]
Thanks, I'll check out `GrammarLocation` and get back if it does not fit my use case.

**Reply by six5536** [reply]
I could achieve everything I needed with `GrammarLocation`
